From: Rafael J. Wysocki <rjw@suse.de>
Subject: x86 / reboot: Blacklist Dell OptiPlex 990 known to require PCI reboot 
Patch-mainline: Not yet
References: bnc#727324

Dell OptiPlex 990 is known to require PCI reboot, so add it to the reboot
blacklist in pci_reboot_dmi_table[].

Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/kernel/reboot.c |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: linux-3.0-SLE11-SP2/arch/x86/kernel/reboot.c
===================================================================
--- linux-3.0-SLE11-SP2.orig/arch/x86/kernel/reboot.c
+++ linux-3.0-SLE11-SP2/arch/x86/kernel/reboot.c
@@ -447,6 +447,14 @@ static struct dmi_system_id __initdata p
 			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E6420"),
 		},
 	},
+	{	/* Handle problems with rebooting on the OptiPlex 990. */
+		.callback = set_pci_reboot,
+		.ident = "Dell OptiPlex 990",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "OptiPlex 990"),
+		},
+	},
 	{ }
 };
 
